### DISCUSSION TITLE
What I am doing wrong here?

### DISCUSSION THREAD

**Discussion by initeshjain** [community/community#26314] [Actions]
<p><img src="https://user-images.githubusercontent.com/47496910/181096862-c1ef320e-3398-4247-b541-a41d31aef0e4.png" alt="image" data-base62-sha1="n1Dj3XdzpCxpTAeM3GlKXaKXd3q" width="490" height="155"><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/d/cd93260a45b8088890b4d4c0e27d9ef63505965e.png" data-download-href="/uploads/short-url/tkB92EqUgyemlDYEBbEylfX2moS.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47496910/181096863-14db9e0b-de1c-4e65-a8d3-2f39ea1bdc9c.png" alt="image" data-base62-sha1="tkB92EqUgyemlDYEBbEylfX2moS" width="690" height="64" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd93260a45b8088890b4d4c0e27d9ef63505965e_2_690x64.png, https://aws1.discourse-cdn.com/github/original/3X/c/d/cd93260a45b8088890b4d4c0e27d9ef63505965e.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/d/cd93260a45b8088890b4d4c0e27d9ef63505965e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd93260a45b8088890b4d4c0e27d9ef63505965e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">724×68 17.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>CODE:</p>
<ul>
<li>name: If intention is to rollback<br>
if: ${{ github.event.inputs.intention == ‘rollback’ }}<br>
run: |<br>
echo “artifactname=${{ env.rollbackartifact }}” &gt;&gt; $GITHUB_ENV<br>
echo ${env:artifactname}</li>
</ul>
<p>I am getting a bad substitution error, please help me. Is there anything wrong with this code?</p>
<p>background:<br>
<strong>env.rollbackartifact</strong> is I am getting from an input field.</p>

**Comment by airtower-luna** [comment, marked as answer]
<pre><code class="lang-yaml">run: |
  echo "artifactname=${{ env.rollbackartifact }}" &gt;&gt; $GITHUB_ENV
  echo ${env:artifactname}
</code></pre>
<p>There are two problems here:</p>
<ol>
<li>You try to set an environment variable variable via <code>GITHUB_ENV</code> and use it in the same run step. That doesn’t work, writing to <code>GITHUB_ENV</code> affects <em>following</em> steps within the job.</li>
<li>
<code>${env:artifactname}</code> isn’t valid Unix shell syntax. If you want to use the <code>artifactname</code> environment variable, the correct code is <code>${artifactname}</code>.</li>
</ol>

**Comment by initeshjain** [comment]
<p>Thanks for the solution.<br>
I just removed the second line from Run step and it worked.</p>
